### ISSUE TITLE
Cache created in feature branch not usable in the default branch (`main`)

### ISSUE THREAD

**Issue by garysassano** [actions/cache#1496]
These two sentences say the opposite:

> Workflow runs can restore caches created in either the current branch or the default branch (usually `main`).

> When a cache is created by a workflow run triggered on a pull request, the cache is created for the merge ref (`refs/pull/.../merge`). Because of this, the cache will have a limited scope and can only be restored by re-runs of the pull request. It cannot be restored by the base branch or other pull requests targeting that base branch.

Source: [Restrictions for accessing a cache](https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/caching-dependencies-to-speed-up-workflows#restrictions-for-accessing-a-cache)

The default branch (`main`) should have access to the cache created in the feature branch that was merged into it.

**Comment by DanRStevens** [comment]
I'm not involved with this project, but that was something interesting to think about. I'm not sure there's necessarily a contradiction there.

A pull request run creates a merge commit, but that merge commit isn't on any branch. It's only after the pull request is merged that any merge commit would become part of the base branch. There's often a significant time gap between when a pull request workflow run is done and when the pull request has tests pass, is reviewed, and is merged.

Additionally, when the pull request is merged, it may get a different merge commit than was used by the pull request workflow run, since there may have been additional changes added to the base branch before the pull request was actually merged.

Even if no changes were added to the base branch between the run and the merge, I suspect the actual merge would still generate a different merge commit with a different SHA, due to different meta data, such as committer/author date.

With that said, this issue being opened is perhaps evidence the documentation could be improved on how pull request runs operate.

**Comment by joshmgross** [comment]
I'm not seeing the contradiction in those sentences, can you clarify your question @garysassano?

> The default branch (`main`) should have access to the cache created in the feature branch that was merged into it.

This is an intentional security protection of the cache to prevent workflows running on feature branches from poisoning the default branch cache.

**Comment by joshmgross** [comment]
Feel free to reopen if I didn't answer your question.

I also created #1549 to gather feedback on the current ref restrictions for caches.
